### PR TITLE
Combine Filters Reproduction into new e2e job

### LINKED ISSUE #43345: Split up filter e2e jobs

### PR DESCRIPTION
**Pull request by iethree**
resolves https://github.com/metabase/metabase/issues/43345

This improves e2e CI performance by:
- increasing parallelization of some of our slowest e2e suites
- combining lots of little test files into a couple larger ones to reduce test setup overhead

spec | time
-- | --
Filters | 28 min ➡️  21 min
Dashboard-filters | 27 min ➡️  18 min
filter-reproductions (new) | 9 min

### CODE REVIEW

**Comment by iethree** [comment]
waiting until #41975  merges to merge this one

**Review by uladzimirdev — APPROVED** [review]
there are a few conflicts to resolve, otherwise LGTM

**Pull request merged**
